### PR TITLE
Enable Navi32 in CI

### PR DESCRIPTION
**Pull request by TedThemistokleous**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -134,6 +134,13 @@ rocmtest clang_debug: rocmnode('vega') { cmake_build ->
 //        cmake_build(flags: "-DCMAKE_BUILD_TYPE=release")
 //    }
 //}
+, clang_release_navi: rocmnode('navi32') { cmake_build ->
```
[the remark this answers is not in the document]
Comment (TedThemistokleous) [reply Jenkinsfile]: Done

**Review by pfultz2 — APPROVED** [review]

**Comment by causten** [comment]
I'm looking at the build logs and the navi32 job does not appear to be running on Navi32 hardware....

http://micimaster.amd.com/blue/organizations/jenkins/MLLibs%2FAMDMIGraphX/detail/enable_navi_32_ci/3/pipeline/52

[2023-05-31T21:46:31.113Z] HOSTNAME=f08519e0014d
[2023-05-31T21:46:31.113Z] RUN_CHANGES_DISPLAY_URL=http://micimaster.amd.com/job/MLLibs/job/AMDMIGraphX/job/enable_navi_32_ci/3/display/redirect?page=changes
[2023-05-31T21:46:31.113Z] dbsshuser=miopenpdb
[2023-05-31T21:46:31.113Z] HUDSON_URL=http://micimaster.amd.com/
[2023-05-31T21:46:31.113Z] jfrog_user=rocm-smi
[2023-05-31T21:46:31.113Z] NODE_LABELS=austin gfx90a miopen rocm-framework-30 rocmtest


@TedThemistokleous

File: Jenkinsfile [hunk 2]
```diff
@@ -140,11 +140,13 @@ rocmtest clang_debug: rocmnode('vega') { cmake_build ->
         def debug_flags = "-g -O2 -fno-omit-frame-pointer -fsanitize=${sanitizers} -fno-sanitize-recover=${sanitizers}"
         cmake_build(flags: "-DCMAKE_BUILD_TYPE=debug -DMIGRAPHX_ENABLE_PYTHON=Off -DMIGRAPHX_ENABLE_GPU=Off -DMIGRAPHX_ENABLE_CPU=On -DCMAKE_CXX_FLAGS_DEBUG='${debug_flags}' -DCMAKE_C_FLAGS_DEBUG='${debug_flags}'")
     }
-}//, clang_release_navi: rocmnode('navi21') { cmake_build ->
-//    stage('HIP Clang Release Navi') {
-//        cmake_build(flags: "-DCMAKE_BUILD_TYPE=release")
-//    }
-//}
+}, clang_release_navi: rocmnode('navi32') { cmake_build ->
```
Comment (pfultz2) [inline Jenkinsfile:143]: The `rocmnodename` function above needs to be update to add the labels for navi32.

Comment (TedThemistokleous) [reply Jenkinsfile:145]: done

**Comment by TedThemistokleous** [comment]
Currently speaking with Omkar, it looks like the Navi32 nodes are down for now. Renamed the node to use gfx1101. Also added debug pipeline

**Review by causten — APPROVED** [review]

File: Jenkinsfile [hunk 3]
```diff
@@ -144,11 +146,17 @@ rocmtest clang_debug: rocmnode('cdna') { cmake_build ->
...
+    stage('HIP Clang Release Navi32') {
+        cmake_build(flags: "-DCMAKE_BUILD_TYPE=release")
+    }
+}, clang_release_navi_debug: rocmnode('navi32') { cmake_build ->
```
Comment (pfultz2) [inline Jenkinsfile:153]: We should only add the release version. I dont think we need to have a debug version enabled in the CI.

Comment (TedThemistokleous) [reply Jenkinsfile:153]: Okay. I'll remove that

**Comment by TedThemistokleous** [comment]
we seem to be failing a few tests in CI right now. Looking into this. Not sure if  there is something missing on our end or the ROCm stack for 5.6

**Comment by TedThemistokleous** [comment]
Seeing this currently on our gpu_async test via make_check()

image

**Comment by TedThemistokleous** [comment]
Trying this as ROCm 5.7 changes have gotten into develop. Hoping this resolves CI issues we were seeing

**Comment by causten** [comment]
Add  -DGPU_TARGETS='${gpu_targets}' to the cmake_build

**Comment by TedThemistokleous** [comment]
Reran the build seeing a few fails 

```
[ COMPLETE ] gemm_2args_mm_4
[   RUN    ] gemm_2args_mm_5
Benchmarking gpu::mlir_op: 21 configs
Fastest solution: 64,32,32,4,2,2
FAILED: gpu
RMS Error: 0.504296
Max diff: 1.61328
Mismatch at 24: 0.121094 != -0.171875

328/328 Test #327: test_py_3.8_backend .......................................................***Timeout 5000.10 sec
```

**Comment by TedThemistokleous** [comment]
Running this on our navi32 rack, ctr-smc-s29-040 I'm still running after a make check. 

it appears a few of our tests are timing out and failing and its been running for the last few hours. Dump below. Will take a look at the test_gpu_fuse_mlir failure. I sense this may be the root of this.

```
315/337 Test #320: test_verify_general .......................................................***Timeout 5000.10 sec
[   RUN    ] batch_quant_dot_1
MIGraphX verification passed successfully.

        Start 321: test_verify_rnn
316/337 Test #319: test_api_custom_op_gpu ....................................................***Timeout 4999.99 sec
[   RUN    ] register_half_copy_op
[ COMPLETE ] register_half_copy_op
[   RUN    ] half_copy_custom_op_test

317/337 Test #318: test_api_gpu ..............................................................***Timeout 4999.99 sec
[   RUN    ] load_and_run

318/337 Test #333: test_py_3.8_gpu_offload ...................................................***Timeout 4999.99 sec

319/337 Test #334: test_py_3.8_gpu ...........................................................***Timeout 4999.99 sec

320/337 Test #335: test_py_3.8_array .........................................................***Timeout 4999.99 sec

321/337 Test #337: test_py_3.8_gpu_async .....................................................***Timeout 4999.99 sec

322/337 Test #336: test_py_3.8_backend .......................................................***Timeout 5000.00 sec

323/337 Test #321: test_verify_rnn ...........................................................***Timeout 5000.10 sec
[   RUN    ] test_gru_bidirct
WARNING: Instruction needs normalization, performance may be affected.
WARNING: Instruction needs normalization, performance may be affected.
MIGraphX verification passed successfully.
MIGraphX verification passed successfully.

        Start  73: test_gpu_adjust_allocation
324/337 Test  #73: test_gpu_adjust_allocation ................................................   Passed    0.13 sec
        Start  74: test_gpu_codegen_literal
325/337 Test  #74: test_gpu_codegen_literal ..................................................***Timeout 5000.10 sec
[   RUN    ] mul_literal_round_test

        Start  75: test_gpu_context_serialize
326/337 Test  #75: test_gpu_context_serialize ................................................***Timeout 5000.10 sec
[   RUN    ] gpu_context_serialize
[ COMPLETE ] gpu_context_serialize
[   RUN    ] context_queue

        Start  76: test_gpu_fuse_mlir
327/337 Test  #76: test_gpu_fuse_mlir ........................................................***Failed    0.16 sec
[   RUN    ] dot_add
terminate called after throwing an instance of 'migraphx::version_1::exception'
  what():  /code/AMDMIGraphX/src/include/migraphx/ranges.hpp:112: at: Operator not found: gpu::mlir_op
CMake Error at gdb/test_test_gpu_fuse_mlir/run.cmake:16 (message):
  Test failed



        Start  77: test_gpu_fuse_ops
328/337 Test  #77: test_gpu_fuse_ops .........................................................   Passed    0.05 sec
        Start  78: test_gpu_hip
```

**Comment by causten** [comment]
Timeouts like that indicate the driver reset the GPU so we are waiting on jobs that no longer exist.  That feels like a bug that we should open against the kernel team.  We should be notified the job has failed so MIGraphX doesn't continually hang.  Now as for the why the job died that's another story and it would be good to get with @okakarpa internally

**Comment by TedThemistokleous** [comment]
> Timeouts like that indicate the driver reset the GPU so we are waiting on jobs that no longer exist. That feels like a bug that we should open against the kernel team. We should be notified the job has failed so MIGraphX doesn't continually hang. Now as for the why the job died that's another story and it would be good to get with @okakarpa internally

Alright, I'll look into it.

Dmesg is showing me 

```
[35706.628805] amdgpu 0000:e3:02.0: amdgpu: [gfxhub] page fault (src_id:0 ring:153 vmid:0 pasid:0, for process  pid 0 thread  pid 0)
[35706.630489] amdgpu 0000:e3:02.0: amdgpu:   in page starting at address 0x00007f1bf0949000 from client 10
```

not sure if thats from us or not though.

**Comment by causten** [comment]
I cant merge this until the failures are resolved.  You can recreate this by manually running the tests

```
cd AMDMIGraphX/build
MIGRAPHX_TRACE_BENCHMARKING=1 ctest -VV -R test_py_3.8_backend .
```

Additionally @umangyadav  wrote (your top folder may vary) ... 
It can be run like `export PYTHONPATH=/build/dir/lib` and then `cd test/onnx && python3 ../py/onnx_backend.py` . In order to print test names, modify onnx_backend.py's  last line to unittest.main(verbosity=2).  There are also other options that unittest module supports.

**Comment by causten** [comment]
By running 1 job at a time I do see see that it ran, but took 6469 seconds.  That's 107 minutes.  This was run on a Navi31 Ubuntu 20.04
`
332/340 Test #332: test_py_3.9_gpu_offload ...................................................   Passed   12.82 sec
        Start 333: test_py_3.8_gpu
333/340 Test #333: test_py_3.8_gpu ...........................................................   Passed   66.26 sec
        Start 334: test_py_3.9_gpu
334/340 Test #334: test_py_3.9_gpu ...........................................................   Passed   65.56 sec
        Start 335: test_py_3.8_array
335/340 Test #335: test_py_3.8_array .........................................................   Passed   12.81 sec
        Start 336: test_py_3.9_array
336/340 Test #336: test_py_3.9_array .........................................................   Passed   12.84 sec
        Start 337: test_py_3.8_backend
337/340 Test #337: test_py_3.8_backend .......................................................   Passed  2895.67 sec
        Start 338: test_py_3.9_backend
338/340 Test #338: test_py_3.9_backend .......................................................   Passed  2846.79 sec
        Start 339: test_py_3.8_gpu_async
339/340 Test #339: test_py_3.8_gpu_async .....................................................   Passed   12.97 sec
        Start 340: test_py_3.9_gpu_async
340/340 Test #340: test_py_3.9_gpu_async .....................................................   Passed   13.02 sec

100% tests passed, 0 tests failed out of 340

Total Test time (real) = 6469.90 sec
root@0385a9397dee:/workspace/AMDMIGraphX/build# /usr/bin/ctest --output-on-failure -j 1 -C . --timeout 5000
`

File: Dockerfile [hunk 4]
```diff
@@ -10,7 +10,7 @@ RUN apt-get update && apt-get install -y gnupg2 --no-install-recommends curl &&
     curl -sL http://repo.radeon.com/rocm/rocm.gpg.key | apt-key add -
 
 # Add rocm repository
-RUN sh -c 'echo deb [arch=amd64 trusted=yes] http://repo.radeon.com/rocm/apt/5.7/ focal main > /etc/apt/sources.list.d/rocm.list'
+RUN sh -c 'echo deb [arch=amd64 trusted=yes] http://repo.radeon.com/rocm/apt/6.0/ focal main > /etc/apt/sources.list.d/rocm.list'
```
Comment (pfultz2) [inline Dockerfile:13]: The version of rocm needs to be updated in all docker files.

Comment (pfultz2) [reply Dockerfile:13]: Also, I think a unit test wont pass with 6.0 due to a bug in hiprtc. I think this needs to use 6.0.1.

**Pull request merged**